File: Tanks/Game/UserTankController.cs
using SharpDX.Mathematics.Interop;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tanks.Game
{
	class UserTankController : ITankController
	{
		public void GetTexture(Tank tank)
		{
			tank.TextureCoords = ResourceManager.Instance.TanksCoords(tank.Type, tank.Direction, tank.Level, 0);
		}

		public void Update(Tank tank, float elapsedTime)
		{
			Input input = Input.Instance;

			if (input["W"])
			{
				tank.Step(Direction.Top);
			}
			else if (input["D"])
			{
				tank.Step(Direction.Right);
			}
			else if(input["S"])
			{
				tank.Step(Direction.Bottom);
			}
			else if (input["A"])
			{
				tank.Step(Direction.Left);
			}
			else if (input["J"])
			{
				tank.Shot();
			}
		}
	}
}

File: TanksNative/UserTankController.cpp
#include "UserTankController.h"
#include "Tank.h"
#include "ResourceManager.h"
#include "Input.h"


void UserTankController::Update(Tank* const tank, float elapsedTime)
{
	if (Input::IskeyDown(L"W"))
	{
		tank->Step(Direction::Top);
	}
	else if (Input::IskeyDown(L"D"))
	{
		tank->Step(Direction::Right);
	}
	else if (Input::IskeyDown(L"S"))
	{
		tank->Step(Direction::Bottom);
	}
	else if (Input::IskeyDown(L"A"))
	{
		tank->Step(Direction::Left);
	}
	else if (Input::IskeyDown(L"J"))
	{
		tank->Shot();
	}
}

void UserTankController::GetTexture(Tank* const tank)
{
	tank->SetTextureCoords(ResourceManager::GetTankCoords(tank->GetType(), tank->GetDirection(), tank->GetLevel(), 0));
}
